dl/src/main/scala/com/intel/analytics/bigdl/optim/Optimizer.scala

```diff
@@ -71,6 +73,18 @@ abstract class Optimizer[T: ClassTag, D](
     this
   }
 
+  def setValidation(trigger: Trigger, sampleRDD: RDD[Sample[T]],
+      vMethods : Array[ValidationMethod[T]], batchSize: Int)
+  : this.type = {
```

Comment:
And what if this is called for LocalOptimizer? I think we need to override the setValidation methods for both LocalOptimizer and DistriOptimizer, which need to check if the validation dataset is LocalDataSet or DistriDataSet respectively.